### ISSUE TITLE
Port Forwarding Rules, "Add associated filter rule" no longer an option after rule is created

### ISSUE THREAD

**Issue by grilparto** [opnsense/core#6057]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [ ] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [ ] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

Not sure if this is working as intended or not.  When creating a Port Forwarding rule, the 'Filter rule association' drop down box provides the option "Add associated filter rule".  If--by mistake for example--the user selects "None", and Saves the rule, the "Add associated filter rule" is no longer an option in the 'Filter rule association' drop down menu.  the only options are "Pass" or "None" or any other filter rules that may have been created in the past.

**To Reproduce**

1. Create a Port Forwarding rule under Firewall > NAT > Port Forward
2. Select "None" from the 'Filter rule association' drop down
3. Click 'Save' to save the rule.
4. Edit the rule
5. Observe that there is no longer an option to "Add associated filter rule" in the 'Filter rule association' drop down

**Workaround**

Creating a new rule is the only way to correct this.  With that in mind the user can delete the rule, and create it again.  Or clone the rule (which is functionally the same as creating a new one but requires less effort), and delete the old one.

**Expected behavior**

I would expect the option to "Add associated filter rule" be available after creating a Port Forwarding rule.

**Environment**

OPNsense 22.7-amd64
FreeBSD 13.1-RELEASE
OpenSSL 1.1.1q 5 Jul 2022

Hardware isn't important in this case.

**Comment by AdSchellevis** [comment]
It's intended to only offer new rules on initial creation: 

https://github.com/opnsense/core/blob/3b9859e07b3614059dbfd720965485588474f1e8/src/www/firewall_nat_edit.php#L1081-L1097

It wouldn't be too much of an issue to allow creation of a new rule if not linked to anything yet, but may have challenges of its own when someone unlinks a rule and saves it. If someone opens I PR, I don't mind having a look.

**Comment by grilparto** [comment]
> but may have challenges of its own when someone unlinks a rule and saves it.

Yeah, I think that makes the case that it's working as intended.  I can imagine there could be a can of worms trying to add the option for existing rules.  It's relatively easy create the rule again or clone it (now that I know about it. I selected "None" and proceeded to pull my hair out trying to figure out why the Port Forwarding rule wasn't working) to.

A different way to mitigate this might be to clarify it [in the documentation](https://docs.opnsense.org/manual/nat.html#port-forwarding) and catch a few people before they select "None" without understanding the impact.  

It mentions the definition of the function:
> Filter rule association: Associate this with a regular firewall rule.

But it doesn't describe the different options.  The default selection "Add associated filter rule" is likely the one most users will want to select, but it wasn't clear to me at first that this meant, "I'm also going to create a firewall rule for you to make things easy."
